src/wallet/rpcwallet.cpp

```diff
@@ -64,7 +64,10 @@ void WalletTxToJSON(const CWalletTx& wtx, UniValue& entry)
 {
     int confirms = wtx.GetDepthInMainChain();
     bool fLocked = instantsend.IsLockedInstantSendTransaction(wtx.GetHash());
-    bool chainlock = llmq::chainLocksHandler->HasChainLock(wtx.nIndex, wtx.hashBlock);
+    bool chainlock = false;
+    if (confirms > 0) {
+        chainlock = llmq::chainLocksHandler->HasChainLock(mapBlockIndex[wtx.hashBlock]->nHeight, wtx.hashBlock);
```

Comment:
this adds a dependency on `cs_main` being held. I verified that it's held by all callers, but we should maybe add a `AssertLockHeld` line.

Reply:
Good point. `mapBlockIndex` is used below too though, so it's not a new dependency but asserting here won't hurt I guess.

Reply:
can you move the AssertLockHeld to the top of the method? I find it easier to spot as it's on the top in most (nearly all) cases. Also, if a method has conditional dependency on a lock, it should be treated the same as if it has this dependency all the time (otherwise we'll get a lot of fun debugging issues that only happen in very special situations).

Reply:
done